### PR TITLE
Fix floating-point underflow during root finding in cvode and arkode

### PR DESCRIPTION
**Pull request by avandecreme**
This is a PR cherry picking https://github.com/novadiscovery/sundials/commit/f7a725e9964a4177b6055a972ff25fea48c404f0 to fix issue #57

### CODE REVIEW

**Review by drreynolds — CHANGES_REQUESTED** [review]
I think that these changes are great.  However, I have some additional requests before this contribution can be accepted:
1. Please add a sentence to the top-level `CHANGELOG.md` indicating the fix.  Also add a similar comment to the "Changes from previous versions" section in both the files `doc/arkode/guide/source/Introduction.rst` and `doc/cvode/guide/source/Introduction.rst`.
2. Request to merge the PR into the `develop` branch of the main SUNDIALS repository (not `main`)

**Comment by avandecreme** [comment]
> I think that these changes are great. However, I have some additional requests before this contribution can be accepted:
> 
>     1. Please add a sentence to the top-level `CHANGELOG.md` indicating the fix.  Also add a similar comment to the "Changes from previous versions" section in both the files `doc/arkode/guide/source/Introduction.rst` and `doc/cvode/guide/source/Introduction.rst`.
> 
>     2. Request to merge the PR into the `develop` branch of the main SUNDIALS repository (not `main`)

Thanks for reviewing. I did the changes you asked and assumed the next version would be 6.4.2 but let me know if I should put something else.

File: doc/arkode/guide/source/Introduction.rst [hunk 1]
```diff
@@ -118,6 +118,11 @@ provided with SUNDIALS, or again may utilize a user-supplied module.
 Changes from previous versions
 ==============================
 
+Changes in v6.4.2
```
Comment (drreynolds) [inline doc/arkode/guide/source/Introduction.rst:121]:
```suggestion
Changes in v5.4.2
```

**Review by drreynolds — DISMISSED** [review]

**Review by balos1 — DISMISSED** [review]
LGTM

**Comment by balos1** [comment]
@gardner48 has pointed out offline that we will want to make this change in the other integrator packages as well.

**Review by drreynolds — APPROVED** [review]

**Review by balos1 — DISMISSED** [review]

**Comment by balos1** [comment]
Thanks for this @avandecreme, it is very appreciated. One last request, could you please sign the commit as indicated at the bottom of the [contributing guide](https://github.com/LLNL/sundials/blob/main/CONTRIBUTING.md)?

**Review by gardner48 — DISMISSED** [review]

**Comment by avandecreme** [comment]
Signed-off added.

**Review by drreynolds — APPROVED** [review]

**Review by gardner48 — APPROVED** [review]

**Pull request merged**